include/swift/AST/TypeRepr.h

```diff
@@ -363,55 +373,20 @@ class DeclRefTypeRepr : public TypeRepr {
 ///   Bar<Gen>
 /// \endcode
 class IdentTypeRepr : public DeclRefTypeRepr {
```

Comment:
That’s the plan! With this PR, there are around  9 `TypeKind` checks and dynamic casts benefitting from the type-level distinction; no API-level clients apart from visitors. I’m not entirely happy about having to call `hasGenericArgList` on top of `isa`, but merging `SimpleIdentTypeRepr` and `GenericIdentTypeRepr` also makes `isa` faster for `IdentTypeRepr` and `DeclRefTypeRepr`, and it feels like it will balance out.

Reply:
#70608.